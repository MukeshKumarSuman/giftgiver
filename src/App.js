import React, {Component } from 'react';
import './App.css';
import {Button} from 'react-bootstrap';
import Gift from './Gift';
import { maxNumber } from './helper';

class App extends Component {
  state = {
    gifts: []
  }

 addGift = event => {
  const { gifts } = this.state;
  const ids = this.state.gifts.map(gift => gift.id);
  const maxId = maxNumber(ids);
  gifts.push({id: maxId + 1});
  this.setState(gifts);
 } 

 removeGift = id => {
  const gifts = this.state.gifts.filter(gift => gift.id !== id);
  this.setState({gifts});
 }



 render() {
  return (
    <div className="App">
      <h1>Gift Givere</h1>
      <div className='gift-list'>
        {this.state.gifts.map(gift => <Gift key={gift.id} gift={gift} removeGift={this.removeGift}/>)}
      </div>
      <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
    </div>
  );
 }
}
export default App;
