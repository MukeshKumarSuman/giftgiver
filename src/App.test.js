import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

import {createSerializer} from 'enzyme-to-json';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('App Module', () => {
  const app = shallow(<App />);

  it('Render properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('initializes the state with an empty list of gift', () => {
    expect( app.state().gifts).toEqual([]);
  });

  describe('when cliking the add gift button', () => {

    beforeEach(() => {
      app.find('.btn-add').simulate('click');
    });

    // reverse the effect of beforeEach
    afterEach(() => {
      app.setState({gifts: []});
    });

    it('add a new gift to state', () => {
      expect( app.state().gifts).toEqual([{id: 1}]);
    });
  
    it('add a new gift to the render list', () => {
      expect( app.find('.gift-list').children().length).toEqual(1);
    });
  });

  describe('when cliking the remove gift button', () => {
    const id = 1;
    beforeEach(() => {
      app.find('.btn-add').simulate('click');
      app.instance().removeGift(id);
    });

    it('remove the gift from state', () => {
      expect(app.state().gifts).toEqual([]);
    });

  });

});
