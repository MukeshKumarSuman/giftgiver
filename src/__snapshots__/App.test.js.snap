// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Module Render properly 1`] = `
<div
  className="App"
>
  <h1>
    Gift Givere
  </h1>
  <div
    className="gift-list"
  />
  <Button
    active={false}
    className="btn-add"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Add Gift
  </Button>
</div>
`;
